cooked-validators/src/Cooked/MockChain/BlockChain.hs

```diff
@@ -278,6 +274,55 @@ valueFromTxOutRef oref = do
     Nothing -> return Nothing
     Just out -> return . Just $ outputValue out
 
+-- ** Slot and Time Management
+
+-- $slotandtime
+-- #slotandtime#
+--
+-- Slots are integers that monotonically increase and model the passage of time. By looking
+-- at the current slot, a validator gets to know that it is being executed within a certain
+-- window of wall-clock time. Things can get annoying pretty fast when trying to mock traces
+-- and trying to exercise certain branches of certain validators; make sure you also read
+-- the docs on 'autoSlotIncrease' to be able to simulate sending transactions in parallel.
+
+-- | Moves n slots fowards
+waitNSlots :: (MonadBlockChainWithoutValidation m) => Integer -> m Ledger.Slot
+waitNSlots n =
+  if n < 0
+    then fail "waitNSlots: negative argument"
+    else currentSlot >>= awaitSlot . (+ fromIntegral n)
+
+-- | Returns the ms interval corresponding to the current slot
+currentTime :: (MonadBlockChainWithoutValidation m) => m (PV2.POSIXTime, PV2.POSIXTime)
+currentTime = do
+  slotConfig <- Emulator.pSlotConfig <$> getParams
+  interval <- Emulator.slotToPOSIXTimeRange slotConfig <$> currentSlot
+  case interval of
+    PV2.Interval (PV2.LowerBound (PV2.Finite l) _) (PV2.UpperBound (PV2.Finite r) _) ->
+      return (l, r)
+    _ -> fail "The enclosing slot should be finite on both ends."
+
+getEnclosingSlot :: (MonadBlockChainWithoutValidation m) => PV2.POSIXTime -> m Ledger.Slot
+getEnclosingSlot t = do
+  slotConfig <- Emulator.pSlotConfig <$> getParams
+  return $ Emulator.posixTimeToEnclosingSlot slotConfig t
+
+-- | Attempts to wait until the slot containing the given time or returns the current slot.
+--  Note that that it might not wait for anything if the resulting slot
+--  is larger than the argument.
```

Comment:
```suggestion
-- | Waits until the current slot becomes greater or equal to the slot containing the given POSIX time.
--  Note that that it might not wait for anything if the current slot is large enough.
```